frontend/packages/console-app/src/components/cloud-shell/AutoResizeTerminal.tsx

```diff
@@ -0,0 +1,139 @@
+  onData: any;
+  className?: string;
+};
+
+type TerminalState = {
+  width: number;
+  height: number;
+};
+
+export class Terminal extends React.Component<TerminalProps, TerminalState> {
+  private innerRef;
+
+  private outerRef;
+
+  private onDataReceived;
+
+  private terminal;
+
+  private padding;
+
+  constructor(props) {
+    super(props);
+    this.innerRef = React.createRef();
+    this.outerRef = React.createRef();
+    this.state = {
+      width: 0,
+      height: 0,
+    };
+    this.onDataReceived = (data) => {
+      this.terminal && this.terminal.write(data);
+    };
+    this.padding = 20;
+    const options = {
+      fontFamily: 'monospace',
+      fontSize: 16,
+      cursorBlink: false,
+      cols: 80,
+      rows: 25,
+    };
+    this.terminal = new XTerminal(Object.assign({}, options));
+    this.terminal.on('data', this.props.onData);
+  }
+
+  focus = () => {
+    this.terminal && this.terminal.focus();
+  };
+
+  onResize = () => {
+    const node = this.outerRef.current;
+
+    if (!node) {
+      return;
+    }
+
+    const bodyRect = document.body.getBoundingClientRect();
+    const nodeRect = node.getBoundingClientRect();
+
+    // This assumes we want to fill everything below and to the right.  In full-screen, fill entire viewport
+    // Use body height when node top is too close to pageRect height
+    const { bottom } = bodyRect;
+    const height = Math.floor(bottom - nodeRect.top);
+    const width = Math.floor(bodyRect.width - nodeRect.left);
+
+    if (height === this.state.height && width === this.state.width) {
+      return;
+    }
+    // rerender with correct dimensions
+    this.setState({ height, width }, () => {
+      const { terminal } = this;
+      if (!terminal) {
+        return;
+      }
+      // tell the terminal to resize itself
+      terminal.fit();
+    });
+  };
+
+  onConnectionClosed = (reason) => {
+    const { terminal } = this;
+    if (!terminal) {
+      return;
+    }
+    terminal.write(`\x1b[31m${reason || 'disconnected'}\x1b[m\r\n`);
+    terminal.cursorHidden = true;
+    terminal.setOption('disableStdin', true);
+    terminal.refresh(terminal.y, terminal.y);
+  };
+
+  componentDidMount() {
+    this.terminal.open(this.innerRef.current);
+    this.focus();
+    this.onResize();
+    this.onDataReceived();
+  }
+
+  componentWillUnmount() {
+    this.terminal && this.terminal.destroy();
+  }
+
+  render() {
+    return (
+      <Measure
+        bounds
+        onResize={() => {
+          this.onResize();
+        }}
+      >
+        {({ measureRef }) => (
+          <div
+            style={{
+              width: '100%',
+              height: '100%',
+              padding: this.padding / 2,
+            }}
+            ref={measureRef}
+          >
+            <div ref={this.outerRef} className={this.props.className}>
+              <div
+                ref={this.innerRef}
+                style={{ width: this.state.width, height: this.state.height }}
```

Comment:
You shouldn't need to set the width and height here.

Reply:
removed